controller/state.go

```diff
@@ -802,18 +818,25 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *v1
 
 // useDiffCache will determine if the diff should be calculated based
 // on the existing live state cache or not.
-func useDiffCache(noCache bool, manifestInfos []*apiclient.ManifestResponse, sources []v1alpha1.ApplicationSource, app *v1alpha1.Application, manifestRevisions []string, statusRefreshTimeout time.Duration, log *log.Entry) bool {
+func useDiffCache(noCache bool, manifestInfos []*apiclient.ManifestResponse, sources []v1alpha1.ApplicationSource, app *v1alpha1.Application, manifestRevisions []string, statusRefreshTimeout time.Duration, serverSideDiff bool, log *log.Entry) bool {
 
 	if noCache {
 		log.WithField("useDiffCache", "false").Debug("noCache is true")
 		return false
 	}
-	_, refreshRequested := app.IsRefreshRequested()
+	refreshType, refreshRequested := app.IsRefreshRequested()
 	if refreshRequested {
-		log.WithField("useDiffCache", "false").Debug("refreshRequested")
-		return false
+		switch refreshType {
+		case v1alpha1.RefreshTypeNormal:
+			log.WithField("useDiffCache", "false").Debug("normal refresh Requested")
+			return false
+		case v1alpha1.RefreshTypeHard:
+			log.WithField("useDiffCache", "false").Debug("hard refresh requested")
+			return false
+		}
 	}
-	if app.Status.Expired(statusRefreshTimeout) {
+	// serverSideDiff should still use cache even if status is expired
```

Comment:
Why is that the case?

Reply:
I am not too sold on that tbh. My intention was relying a bit more on the cache when ssd is enabled to avoid having to do too many calls to kubeapi.

Reply:
That's fair. I think it's fine to go that route, as long as the comment explains the decision and the potential downsides.

Reply:
added